MantidQt/CustomInterfaces/src/Reflectometry/QtReflMainView.cpp

```diff
@@ -86,163 +83,66 @@ void QtReflMainView::setTransferMethods(const std::set<std::string> &methods) {
-    m_openMap->setMapping(openTable, QString::fromStdString(*it));
-    // When repeated corrections happen the QMessageBox from openTable()
-    // method in ReflMainViewPresenter will be called multiple times
-    // when 'no' is clicked.
-    // ConnectionType = UniqueConnection ensures that
-    // each object has only one of these signals.
-    connect(openTable, SIGNAL(triggered()), m_openMap, SLOT(map()),
-            Qt::UniqueConnection);
-    connect(m_openMap, SIGNAL(mapped(QString)), this, SLOT(setModel(QString)),
-            Qt::UniqueConnection);
+  for (auto it = instruments.begin(); it != instruments.end(); ++it) {
+    QString instrument = QString::fromStdString(*it);
+    ui.comboSearchInstrument->addItem(instrument);
   }
-}
-
-void QtReflMainView::icatSearchComplete() {
-  m_presenter->notify(IReflPresenter::ICATSearchCompleteFlag);
-}
-
-/**
-This slot notifies the presenter that the "save" button has been pressed
-*/
-void QtReflMainView::on_actionSaveTable_triggered() {
-  m_presenter->notify(IReflPresenter::SaveFlag);
-}
-
-/**
-This slot notifies the presenter that the "save as" button has been pressed
-*/
-void QtReflMainView::on_actionSaveTableAs_triggered() {
-  m_presenter->notify(IReflPresenter::SaveAsFlag);
-}
-
-/**
-This slot notifies the presenter that the "append row" button has been pressed
-*/
-void QtReflMainView::on_actionAppendRow_triggered() {
-  m_presenter->notify(IReflPresenter::AppendRowFlag);
-}
-
-/**
-This slot notifies the presenter that the "prepend row" button has been pressed
-*/
-void QtReflMainView::on_actionPrependRow_triggered() {
-  m_presenter->notify(IReflPresenter::PrependRowFlag);
-}
-
-/**
-This slot notifies the presenter that the "delete" button has been pressed
-*/
-void QtReflMainView::on_actionDeleteRow_triggered() {
-  m_presenter->notify(IReflPresenter::DeleteRowFlag);
-}
 
-/**
-This slot notifies the presenter that the "process" button has been pressed
-*/
-void QtReflMainView::on_actionProcess_triggered() {
-  m_presenter->notify(IReflPresenter::ProcessFlag);
-}
-
-/**
-This slot notifies the presenter that the "group rows" button has been pressed
-*/
-void QtReflMainView::on_actionGroupRows_triggered() {
-  m_presenter->notify(IReflPresenter::GroupRowsFlag);
-}
-
-/**
-This slot notifies the presenter that the "clear selected" button has been
-pressed
-*/
-void QtReflMainView::on_actionClearSelected_triggered() {
-  m_presenter->notify(IReflPresenter::ClearSelectedFlag);
-}
+  int index = ui.comboSearchInstrument->findData(
+      QString::fromStdString(defaultInstrument), Qt::DisplayRole);
+  ui.comboSearchInstrument->setCurrentIndex(index);
 
-/**
-This slot notifies the presenter that the "copy selection" button has been
-pressed
-*/
-void QtReflMainView::on_actionCopySelected_triggered() {
-  m_presenter->notify(IReflPresenter::CopySelectedFlag);
+  // Finally, populate the combo box in the table view
+  m_tableView->setInstrumentList(instruments, defaultInstrument);
```

Comment:
I've modified this so that `m_tableView` is populated by its presenter: when `ReflMainViewPresenter` creates the the instrument list, it updates its view and additionally acts on `ReflTableViewPresenter` via the interface `IReflTablePresenter` so that the table view gets populated.